JSDemos/Demos/Toolbar/Multiline/Angular/app/app.service.ts

```diff
@@ -0,0 +1,166 @@
+import { Injectable } from '@angular/core';
+
+export class FontSize {
+  size: number;
+
+  text: string;
+}
+
+export class LineHeight {
+  lineHeight: number;
+
+  text: string;
+}
+
+export class FontFamilies {
+  text: string;
+}
+
+export class FontStyle {
+  icon: string;
+
+  style: string;
+
+  hint: string;
+}
+
+export class TextAlign {
+  icon: string;
+
+  alignment: string;
+
+  hint: string;
+}
+
+export class TextAlignExtended extends TextAlign {
+  text: string;
+}
+
+export class ListType {
+  icon: string;
+
+  alignment: string;
+
+  hint: string;
+}
+
+const fontSizes: FontSize[] = [
+  { size: 10, text: '10px' },
+  { size: 12, text: '12px' },
+  { size: 14, text: '14px' },
+  { size: 16, text: '16px' },
+  { size: 18, text: '18px' },
+];
+
+const lineHeights: LineHeight[] = [
+  { lineHeight: 1, text: '1' },
+  { lineHeight: 1.35, text: '1.35' },
+  { lineHeight: 1.5, text: '1.5' },
+  { lineHeight: 2, text: '2' },
+];
+
+const fontFamilies: FontFamilies[] = [
+  { text: 'Arial' },
+  { text: 'Courier New' },
+  { text: 'Georgia' },
+  { text: 'Impact' },
+  { text: 'Lucida Console' },
+  { text: 'Tahoma' },
+  { text: 'Times New Roman' },
+];
+
+const fontStyles: FontStyle[] = [
+  {
+    icon: 'bold',
+    style: 'bold',
+    hint: 'Bold',
+  },
+  {
+    icon: 'italic',
+    style: 'italic',
+    hint: 'Italic',
+  },
+  {
+    icon: 'underline',
+    style: 'underline',
+    hint: 'Underlined',
+  },
+  {
+    icon: 'strike',
+    style: 'strike',
+    hint: 'Strikethrough',
+  },
+];
+
+const textAlignsExtended: TextAlignExtended[] = [
+  {
+    icon: 'alignleft',
+    alignment: 'left',
+    hint: 'Align left',
+    text: 'Align left',
+  },
+  {
+    icon: 'aligncenter',
+    alignment: 'center',
+    hint: 'Center',
+    text: 'Center',
+  },
+  {
+    icon: 'alignright',
+    alignment: 'right',
+    hint: 'Align right',
+    text: 'Align right',
+  },
+  {
+    icon: 'alignjustify',
+    alignment: 'justify',
+    hint: 'Justify',
+    text: 'Justify',
+  },
+];
+
+const listTypes: ListType[] = [
+  {
+    icon: 'orderedlist',
+    alignment: 'orderedlist',
+    hint: 'Ordered',
+  },
+  {
+    icon: 'bulletlist',
+    alignment: 'bulletlist',
+    hint: 'Bullet',
+  },
+];
+
+@Injectable()
+export class Service {
```

Comment:
It will be cool if we rewrite our Angular services more Angular-way.
I mean instead of returning constants from methods, return rxjs observables
That allows our clients easier use our demos in real projects.

But, in all our demos we write Angular services this way now, so maybe this question needs to be fixed as a different task.

[Link to the Teams discussion about this](https://teams.microsoft.com/l/message/19:7a8fe7395a2d47288c29cfc421df6cbf@thread.skype/1663321380547?tenantId=e4d60396-9352-4ae8-b84c-e69244584fa4&groupId=8c31b3e5-5870-46b1-92a7-180d13b282ed&parentMessageId=1663321380547&teamName=DevExtreme&channelName=General&createdTime=1663321380547&allowXTenantAccess=false)

Reply:
The discussion is not over. We decided to use the existing code style.